app/controllers/tasks_controller.rb

```diff
@@ -1,90 +1,66 @@
 class TasksController < ApplicationController
-  # set_task will be called before show, edit, update, or destroy
   before_action :set_task, only: [:show, :edit, :update, :complete, :destroy]
 
-  # GET /tasks
   def index
+    @person = Person.find(params[:person_id])
+  end
+
+  def all_tasks
     @all_tasks = Task.all
-    # render :index
   end
 
-  # GET /tasks/:id
   def show
-    # @task = Task.find(params[:id])
-    render :show_task
   end
 
-  # GET /tasks/new
   def new
     @task = Task.new
-    render :new_task # return an HTML form for creating a new task
+    @person = Person.find(params[:person_id])
+    @all_people = Person.all
   end
 
-  # # GET /tasks/:id/edit  tasks#edit return an HTML form for editing a task
-  # def edit
-  # end
+  def edit
+    @all_people = Person.all
+    @person = Person.find(@task.person_id)
+  end
 
-  # POST /tasks
-  # create a new task
   def create
     # TODO: require task name from the user
     @task = Task.new(task_params)
+    person = Person.find(task_params[:person_id])
 
-    respond_to do |format|
-      if @task.save
-        # format.html { redirect_to @task } # Ash prefers this
-        format.html {redirect_to tasks_path} # required for project requirements
-        # format.json { render :show, status: :created, location: @task }
-      else
-        format.html { render :new_task }
-        # format.json { render json: @task.errors, status: :unprocessable_entity }
-      end
-    end
-  end
 
-  # PATCH/PUT /tasks/:id
-  # update a specific task
-  def update
-    respond_to do |format|
-      if @task.update(task_params)
-        format.html { redirect_to @task }
-        # format.json { render :show, status: :ok, location: @task }
-      else
-        format.html { render :edit, notice: 'Sorry, something went wrong =(' }
-        # format.json { render json: @task.errors, status: :unprocessable_entity }
-      end
+    if @task.save
+      redirect_to person_tasks_path(person)
+    else
+      render :new, notice: 'Sorry, something went wrong =('
     end
   end
 
-  def complete
-    respond_to do |format|
-      if @task.update(completed_at: Time.now)
-        format.html { redirect_to tasks_path }
-      else
-        format.html { redirect_to @task }
-      end
+  def update
+    if params[:complete]
+      @task.change_complete_status(params[:complete])
+    # TODO: fix -- this is a bit awkward if they delete a task from /tasks
+      redirect_to person_tasks_path(params[:person_id])
+    elsif @task.update(task_params)
+      redirect_to @task
+    else
+      render :edit, notice: 'Sorry, something went wrong =('
     end
   end
 
-
-  # # DELETE /tasks/:id
-  # delete a specific task
   def destroy
     @task.destroy
-    respond_to do |format|
-      format.html { redirect_to tasks_path, notice: "\"#{@task.name}\" was deleted." }
-      # format.json { head :no_content }
-    end
+    person_id = @task.person.id
```

Comment:
tiny nitpick, but this could be `person_id = @task.person_id`, this way you don't load the `Person` model associated with this `Task` which requires a trip to the database and instantiating a new object, whereas `.person_id` just accesses an integer already in memory.